Tests/ASUICollectionViewTests.mm

```diff
@@ -138,29 +138,4 @@ - (void)testThatIssuingAnUpdateBeforeInitialReloadIsUnacceptable
   XCTAssertThrowsSpecificNamed([cv insertSections:[NSIndexSet indexSetWithIndex:0]], NSException, NSInternalInconsistencyException);
 }
 
-// If you put reloadData in a batch update, collection view will ignore it and perform the normal
-// update validation i.e. throw an exception if your data source counts changed.
-- (void)testThatPuttingReloadDataInABatchUpdateDoesntWork
-{
-  UICollectionViewLayout *layout = [[UICollectionViewLayout alloc] init];
-  UICollectionView *cv = [[UICollectionView alloc] initWithFrame:CGRectMake(0, 0, 100, 100) collectionViewLayout:layout];
-  id dataSource = [OCMockObject niceMockForProtocol:@protocol(UICollectionViewDataSource)];
-  // Start data source at 1 section, 1 item
-  [[[dataSource stub] andReturnValue:[NSNumber numberWithInteger:1]] numberOfSectionsInCollectionView:cv];
-  [[[dataSource expect] andReturnValue:[NSNumber numberWithInteger:1]] collectionView:cv numberOfItemsInSection:0];
-
-  cv.dataSource = dataSource;
-
-  // Verify initial data.
-  XCTAssertEqual([cv numberOfSections], 1);
-  XCTAssertEqual([cv numberOfItemsInSection:0], 1);
-  [dataSource verify];
-
-  XCTAssertThrows([cv performBatchUpdates:^{
-    // Change data source to 1 section, 2 items
-    [[[dataSource stub] andReturnValue:[NSNumber numberWithInteger:2]] collectionView:cv numberOfItemsInSection:0];
-    [cv reloadData];
-  } completion:nil]);
-}
-
```

Comment:
Note: This test is broken now. It involves intentionally causing UICollectionView to throw an exception, and the view now ends up in such a state that when it deallocates, it still thinks there's an update in progress which isn't allowed.

The test doesn't really do much. It just confirms an edge case in UICollectionView's behavior.